packages/webapp/src/elections/components/participation-card.tsx

```diff
@@ -0,0 +1,287 @@
+import { useState } from "react";
+import dayjs from "dayjs";
+
+import {
+    Button,
+    Container,
+    ElectionParticipationStatus,
+    Form,
+    Heading,
+    Modal,
+    onError,
+    queryMemberByAccountName,
+    Text,
+    useCurrentElection,
+    useCurrentMember,
+    useUALAccount,
+} from "_app";
+// import { generateEncryptionKey } from "_app/eos/secret-publisher";
+// import { setEncryptionPublicKeyTransaction } from "encryption/transactions";
+
+// import { UpdateEncryptionPassword, useEncryptionPassword } from "../hooks";
+import { useEffect } from "react";
+import { getElectionStartDateTime } from "elections/utils";
+import { setElectionParticipation } from "elections/transactions";
+import { useQueryClient } from "react-query";
+
+interface Props {
+    promptSetupEncryptionKey?: boolean;
+}
+
+export const ParticipationCard = ({ promptSetupEncryptionKey }: Props) => {
+    const [ualAccount] = useUALAccount();
+    const { data: currentMember } = useCurrentMember();
+    const { data: election } = useCurrentElection();
+
+    const [
+        showConfirmParticipationModal,
+        setShowConfirmParticipationModal,
+    ] = useState(false);
+    const [
+        showCancelParticipationModal,
+        setShowCancelParticipationModal,
+    ] = useState(false);
+
+    if (!election) {
+        return null;
+    }
+
+    const electionStartDateTime = getElectionStartDateTime(election);
+
+    if (!electionStartDateTime) {
+        return <Text>Error parsing the Election start date.</Text>;
+    }
+
+    const parsedElectionStartDateTime = dayjs(electionStartDateTime);
+    const parsedParticipationTimeLimit = parsedElectionStartDateTime.subtract(
+        24,
+        "hour"
+    );
+    const parsedEstimatedElectionEndDateTime = parsedElectionStartDateTime.add(
+        10, // TODO: is it dynamic?
```

Comment:
I will leave this fixed value with this TODO, so we can adjust and calculate it dynamically based on how many RSVP confirmations we have since I'm not sure how to get it yet.

Reply:
which could be estimated and minimized by querying community size / RSVPs, calculating the number of rounds needed (rounding up by one if we're near a cross-over point), then planning an hour for each round